File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.cpp
class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
        vector<int> res, index;
        
        // Find the indices where 'x' occurs in 'nums'
        for (int i = 0; i < nums.size(); ++i)
            if (nums[i] == x)
                index.push_back(i);
        
        // For each query, find the corresponding index (if it exists)
        for (auto num : queries) {
            if (num <= index.size())
                res.push_back(index[num - 1]);
            else
                res.push_back(-1); // Index out of bounds
        }
        return res;
    }
};

File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.java
class Solution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        int ans[] = new int[queries.length];
        ArrayList<Integer> list = new ArrayList<>();
        for(int i = 0; i < nums.length ; i++){
            if(nums[i] == x){
                list.add(i);
            }
        }
        int countOfx = 0;
        for(int i : nums){
            if(i == x)countOfx++;
        }
        int ansidx = 0;
        int listidx = 0;
        for(int i : queries){
            if(countOfx >= i){
                ans[ansidx++] = list.get(i-1);
            }else{
                ans[ansidx++] = -1;
            }
        }
        return ans;
    }
}